connectors/sources/microsoft_teams.py

```diff
@@ -214,7 +428,22 @@ async def _get(self, absolute_url):
             await self._handle_client_response_error(absolute_url, e)
 
     async def _handle_client_response_error(self, absolute_url, e):
-        if e.status == 403:
+        if e.status == 429 or e.status == 503:
+            response_headers = e.headers or {}
+            updated_response_headers = {
+                key.lower(): value for key, value in response_headers.items()
+            }
+            retry_seconds = int(
+                updated_response_headers.get("retry-after", RETRY_SECONDS)
+            )
+            self._logger.debug(
+                f"Rate limited by Microsoft Teams. Retrying after {retry_seconds} seconds"
+            )
+            await self._sleeps.sleep(retry_seconds)
+            raise ThrottledError(
```

Comment:
I think this retry logic doubling up with the `@retryable` decorator. Is this intended?

Reply:
Can you please elaborate more on this? How it is doubling up the logic? 
I guess this will retry 3 times(i.e. number of RETRY_COUNT)

Reply:
On line 452 the code is being slept for x seconds, and then the error is raised. After raising the error, the `@retryable` decorator will make the code sleep again for `RETRY_INTERVAL` seconds. So it's doubling up in retry time. 

I can see that we will need to respect the `retry-after` value in the response headers, so I don't see an easy workaround. The code will just pause for longer than expected because it will pause twice.

It's okay like this, I just wanted to make sure we're all aware that this is intended.